pkg/skaffold/runner/deployer.go

```diff
@@ -92,6 +92,19 @@ func GetDeployer(ctx context.Context, runCtx *runcontext.RunContext, labeller *l
 				}
 			}
 		}
+		// `helm template` doesn't consistently apply `--namespace` values to rendered resources (see https://github.com/helm/helm/issues/3553).
```

Comment:
My understanding from the helm issue linkedi is that `helm template` intentionally never applies namespaces (my understanding is that we have custom logic that does this, we don't rely on `helm template` to set the namespace)

Reply:
Not really. For instance:
```
helm --kube-context kind-kind template redis-release redis --namespace foobar --repo https://charts.bitnami.com/bitnami
```
Outputs:
```
...
# Source: redis/templates/serviceaccount.yaml
apiVersion: v1
kind: ServiceAccount
automountServiceAccountToken: true
metadata:
  name: redis-release
  namespace: "foobar"
...
```
where namespace `foobar` has been applied by Helm directly.

Reply:
You are correct, I should have changed my comment there. `helm template` never adds a new namespace, it is only if the source template has a parameter for the namespace:
```
{{- if (include "redis.createConfigmap" .)) }}
apiVersion: v1
kind: ConfigMap
metadata:
  name: {{ printf  "%s-configuration" (include "common.names.fullname" .)) }}
  namespace: {{ .Release.Namespace | quote }}
```
The comment might be fine as is, just wanted to state that the `helm` team states there they don't plan on making it add a `namespace` field, I believe it only inserts a value into `.Release.Namespace`